### PR TITLE
Fetch and Display Product List using Commercetools API

### PR DESCRIPTION
**Pull request by valeryiay**
1. Task: [Issue RSS-ECOMM-3_01: Fetch and Display Product List using Commercetools API](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_01.md)
2. Screenshot:
Screenshot 2024-06-03 010217

3. Change description
- [x] API Setup: Set up the commercetools API in your project, following the guidelines provided in the API documentation.
- [x] Fetching Data: Use the appropriate commercetools API endpoints to fetch product data. Handle potential errors in data fetching.
- [x] Data Parsing and Display: Parse the fetched data to extract the necessary product details: name, image, and description. Display this data on the Catalog Product page in a user-friendly format.
4. Change rationale
Changes aim to create a more polished and user-friendly catalog page for an e-commerce application, helping users browse products effortlessly.

### CODE REVIEW

**Review by vetalapo — CHANGES_REQUESTED** [review]
Please figure out why your code doesn't build.
image


When you solve that, I think the PR's going to be good to merge.

**Review by JoMarianek — CHANGES_REQUESTED** [review]
Looks great, good job! The design looks nice.
As Vitalii said, the app is not building, and maybe also merge release branch into feature branch, so it can be safely merged back into release branch later?

File: src/views/ProductsView.vue [hunk 1]
```diff
@@ -1,3 +1,120 @@
...
-    <h1>This is Products page</h1>
+    <v-container class="catalog-container" fluid>
+        <v-row>
+            <v-col cols="12">
+                <h1 class="text-center">Product Catalog</h1>
+            </v-col>
+        </v-row>
+        <v-row>
+            <v-col cols="12" class="text-center">
+                <v-alert v-if="loading" type="text">Loading...</v-alert>
```
Comment (vetalapo) [inline src/views/ProductsView.vue]: There is no type "text" for v-alert.
Your options here like "error" | "success" | "warning" | "info" | undefined'

**Review by vetalapo — CHANGES_REQUESTED** [review]
Please figure out build issues

image

**Review by vetalapo — CHANGES_REQUESTED** [review]
Please resolve build issues:

image

File: src/types/index.ts [hunk 2]
```diff
@@ -130,6 +130,14 @@ type ProductAllData = {
...
 };
 
+interface ProductName {
+    [key: string]: string;
+}
+
+interface ProductDescription {
+    [key: string]: string;
+}
+
```
Comment (vetalapo) [inline src/types/index.ts:140]: Is there really a need to have 2 of the same interfaces but different names?
Maybe you could condense them into some general purpose type? Like StringDictionary or something like that.

**Review by vetalapo — CHANGES_REQUESTED** [review]
We almost have a good PR. Please check a couple of more things.

**Review by vetalapo — APPROVED** [review]

**Review by JoMarianek — APPROVED** [review]

**Pull request merged**